### ISSUE TITLE
Incorrect mineral assignment

### ISSUE THREAD

**Issue by Matt-Power** [Bruker-Amics/bugs_features#295]
V 3.3.0.1315
Segments are mis-identified / spectra do not match at all. For instance, pyrite is identified as quartz even though the spectrum is clearly pyrite and very definitely not quartz - see example below.

Seems to be particularly bad for small segments - it's almost as if a clean up step is being applied that allows small segments to be assigned to the surrounding mineral. However, no clean up steps or spectrum tree steps are included in the processing. Example below. @MarkGuo7 I'll send a link to the file so you can have a look.

Needless to say, this is a really significant issue.

![image](https://user-images.githubusercontent.com/84986921/194374695-211e3b10-4425-46da-a057-5ab52d02d3ce.png)

**Comment by Matt-Power** [comment]
I should add:
This is everywhere in the measurement and is not related to field boundaries.
Not all samples in the batch seem to show it.
Reclassification of the sample does not change things.
It is really difficult to spot the issue, especially when running a large batch of samples - each has to be checked individually.
Reclassifying in 3.2.1.0 does fix the problem i.e. issue lies in newer builds.

**Comment by MarkGuo7** [comment]
@Matt-Power Hi Matt, sorry about that. I will take a look of it today.

**Comment by Matt-Power** [comment]
No worries - all part of the process. Had me scratching my head though. Thank you.

**Comment by MarkGuo7** [comment]
@Matt-Power 

Hi Matt

Take a quick look of your result, seems the touchup step made the difference:
If I classify with this step on:

![image](https://user-images.githubusercontent.com/84481724/194447447-f084cad9-48a6-4ec8-8b59-f97c04a8dbc0.png)

I got this result as you showed me:

![image](https://user-images.githubusercontent.com/84481724/194447483-0d7e7945-e3f7-4b6f-b8f6-1b4da3a92811.png)

If I switched it off:

![image](https://user-images.githubusercontent.com/84481724/194447533-7a2bfc2c-417e-4dcd-9d9d-d0a58907001f.png)

Then I got a different result:

![image](https://user-images.githubusercontent.com/84481724/194447667-2a994140-865c-4408-8509-8c1310276fbd.png)

Could you confirm if it is a classification issue or caused by touchup step?

**Comment by Matt-Power** [comment]
I tried reclassifying without any steps or touchups previously and it made no difference.

Furthermore, that touchup step is simply any unknowns with an area <50 can be changed to anything else. It should not touch anything that is classified already.

**Comment by Matt-Power** [comment]
I should add that this was a random issue too - most other samples in the batch were fine (and the same steps / touchups were applied so I don't think it is simply the touchup step or at least there is something more going on).

**Comment by MarkGuo7** [comment]
@Matt-Power 
Hi Matt
Could you try to classify without this step?
I tried several times, for single classification seems ok. Not sure about batch.

**Comment by Matt-Power** [comment]
I did - several times before I raised the issue.

**Comment by MarkGuo7** [comment]
I will ask Gerda to give it a try see if she can repeat on her computer.

Maybe I need to login your computer to take a look, because I couldn't repeat on mine.

**Comment by Matt-Power** [comment]
Right, just logged back in and in true sod's law style and despite recreating the issue several times before logging it, removing that step (again) worked. Why it didn't work before, I have no idea.

I suppose the issue now comes down to the touchup step. Why is it merging minerals when the step specifies only unknowns are to be merged? Similarly, any idea why that would then persist during reclassification?

**Comment by MarkGuo7** [comment]
@Matt-Power That's cool, thanks.
Please let me know if you notice anything strange happens again.

I will take a look of this touchup issue.

**Comment by Matt-Power** [comment]
This happened again in a different project but with a slightly different result:
Segments surrounding quartz were tagged as Ti oxides (despite there being no Ti anywhere). Removing the touchup step and reclassifying (against the current list, not the ore type) cured the problem.
This was with a similar touchup step - area <50 of unknown can go to any mineral. All segments in the area were classified though so no unknowns present.
So, the issue is definitely in the touchup step handling, not the classification itself.
Unfortunately, I didn't take pics or save but will upload if / when it happens again.

**Comment by MarkGuo7** [comment]
@Matt-Power Thanks Matt

I will take a look of this touchup step issue today.
You last sample would be good enough for me to debugging.

Will let you know if I find anything.

**Comment by MarkGuo7** [comment]
@Matt-Power 

Hi Matt

Could you try this version? https://bruker-my.sharepoint.com/:u:/p/junfeng_guo/EZk3AhlfpexKtUkDfnd6ww0BqOccmMRXpcTEOrTVKg8rHA?e=GwZ1fK

With thanks,

**Comment by Matt-Power** [comment]
Just tried that version but got a blue screen of death / windows stopcode.

It happened when I opened a measurement, loaded the ore types, and went to the grain panel. I created a new touchup step (unknowns area<50 can got to any mineral), got a message saying that it already exists so I clicked to overwrite at which point I got the BSOD. 

Interestingly, no touchup steps were indicated in the reclassify dialog before getting this error even though there was one called "50" which I had created previously and this was confirmed by the message box when I went to the grain panel just before the crash.
